appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/DOLUtils.java

```diff
@@ -873,4 +889,20 @@ public static String getComponentEnvId(JndiNameEnvironment env) {
 
     return id;
   }
+
+/**
+ * Supports extreme classloading isolation
+ *
+ * @param application
+ * @param className
+ * @return true if the class is white-listed
+ */
+    public static boolean isWhiteListed(Application application, String className) {
+        for (String packageName : Iterables.concat(application.getWhitelistPackages(), SYSTEM_PACKAGES)) {
```

Comment:
I've thought long and hard about this, the risk/reward definitely tilts in favor of white-listing system packages.  If someone has Jersey for example, it'll be loaded according to classloading delegation parameter, which is what we want